schema/v1.0/parser.go

```diff
@@ -0,0 +1,111 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema // import "go.opentelemetry.io/otel/schema/v1.0"
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+	"gopkg.in/yaml.v2"
+
+	"go.opentelemetry.io/otel/schema/v1.0/ast"
+)
+
+// Major file version number that this library supports.
+const supportedFormatMajor = 1
+
+// Maximum minor version number that this library supports.
+const supportedFormatMinor = 0
+
+// Maximum major+minor version number that this library supports, as a string.
+var supportedFormatMajorMinor = strconv.Itoa(supportedFormatMajor) + "." +
+	strconv.Itoa(supportedFormatMinor) // 1.0
+
+// ParseFile a schema file. schemaFilePath is the file path.
+func ParseFile(schemaFilePath string) (*ast.Schema, error) {
+	file, err := os.Open(schemaFilePath)
+	if err != nil {
+		return nil, err
+	}
+	return Parse(file)
+}
+
+// Parse a schema file. schemaFileContent is the readable content of the schema file.
+func Parse(schemaFileContent io.Reader) (*ast.Schema, error) {
+	var ts ast.Schema
+	schemaContent, err := ioutil.ReadAll(schemaFileContent)
+	if err != nil {
+		return nil, err
+	}
+
+	err = yaml.Unmarshal(schemaContent, &ts)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := checkFileFormatField(ts.FileFormat); err != nil {
+		return nil, err
+	}
+
+	if strings.TrimSpace(ts.SchemaURL) == "" {
+		return nil, fmt.Errorf("schema_url field is missing")
+	}
+
+	if _, err := url.Parse(ts.SchemaURL); err != nil {
+		return nil, fmt.Errorf("invalid URL specified in schema_url field: %v", err)
```

Comment:
Can we wrap the error to provide more structure the response?

```suggestion
		return nil, fmt.Errorf("invalid URL specified in schema_url field: %w", err)
```